pandas/tests/indexes/test_multi.py

```diff
@@ -1317,6 +1317,18 @@ def test_get_indexer_nearest(self):
         with tm.assertRaises(NotImplementedError):
             midx.get_indexer(['a'], method='pad', tolerance=2)
 
+    def test_hash_collisions(self):
+        # non-smoke test that we don't get hash collisions
+
+        index = MultiIndex.from_product([np.arange(1000), np.arange(1000)],
+                                        names=['one','two'])
+        result = index.get_indexer(index.values)
+        self.assert_numpy_array_equal(result, np.arange(len(index), dtype='int64'))
+
+        for i in [0, 1, len(index)-2, len(index)-1]:
+            result = index.get_loc(index[i])
+            self.assertEqual(result, i)
```

Comment:
yep, that's why this is a non-smoke tests :>

if you have any better ideas on how to test, all ears.